CoreTableView/test/unit/src/au/gov/asd/tac/constellation/views/tableview2/io/TableViewPreferencesIOUtilitiesNGTest.java

```diff
@@ -0,0 +1,214 @@
+    }
+    
+    @Test
+    public void getPreferencesNullPrefs() throws IOException {       
+        jsonIOStaticMock.when(() -> JsonIO.loadJsonPreferences("TableViewPreferences", "vertex-"))
+                .thenReturn(null);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> preferences
+                = TableViewPreferencesIOUtilities.getPreferences(GraphElementType.VERTEX, 2);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> expected = ThreeTuple.create(
+                Collections.emptyList(),
+                Tuple.create("", TableColumn.SortType.ASCENDING),
+                500
+        );
+        
+        assertEquals(expected, preferences);
+    }
+    
+    @Test
+    public void savePreferences() throws IOException {
+        // TODO Find a better solution for this. Because of this limitation these tests
+        //      will not be run on the CI server.
+        if (!GraphicsEnvironment.isHeadless()) {
+            // Interestingly once you throw the skip exception it doesn't call the tear down class
+            // so we need to instantiate the static mocks only once we know we will be running the
+            // tests.
+            new JFXPanel();
+        
+            nbPreferencesStaticMock.when(() -> NbPreferences.forModule(ApplicationPreferenceKeys.class))
+                    .thenReturn(null);
+            applicationPrefKeysStaticMock.when(() -> ApplicationPreferenceKeys.getUserDir(null))
+                    .thenReturn(System.getProperty("java.io.tmpdir"));
+
+            final ObservableList<TableColumn<ObservableList<String>, ? extends Object>> columns = mock(ObservableList.class);
+            when(columns.size()).thenReturn(4);
+
+            final ObservableList<TableColumn<ObservableList<String>, ? extends Object>> sortOrder = mock(ObservableList.class);
+            when(sortOrder.isEmpty()).thenReturn(Boolean.FALSE);
+
+            final TableColumn<ObservableList<String>, ? extends Object> column1 = mock(TableColumn.class);
+            final TableColumn<ObservableList<String>, ? extends Object> column2 = mock(TableColumn.class);
+            final TableColumn<ObservableList<String>, ? extends Object> column3 = mock(TableColumn.class);
+            final TableColumn<ObservableList<String>, ? extends Object> column4 = mock(TableColumn.class);
+
+            when(column1.isVisible()).thenReturn(Boolean.TRUE);
+            when(column1.getText()).thenReturn("ABC");
+
+            when(column2.isVisible()).thenReturn(Boolean.TRUE);
+            when(column2.getText()).thenReturn("DEF");
+            when(column2.getSortType()).thenReturn(TableColumn.SortType.ASCENDING);
+
+            when(column3.isVisible()).thenReturn(Boolean.FALSE);
+            when(column3.getText()).thenReturn("GHI");
+
+            when(column4.isVisible()).thenReturn(Boolean.TRUE); // <- This seem wrong. Its dropped.
```

Comment:
also would be curious to check this out

Reply:
Again. Not sure if the assumption here is that once you hit the first column all columns after will also not be visible. The code does not defend against this being the case nor is there any comment confirming this.

Again need some input from one of the authors.